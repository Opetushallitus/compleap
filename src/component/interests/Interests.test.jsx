import React from 'react'
import createRendererWithTheme from 'test/util/createRendererWithTheme'

const mockState = queryStatus => jest.doMock('state/state', () => {
  const Atom = require('bacon.atom')
  const React = require('react')

  const state = Atom({
    context: {
      interests: {
        data: [{
          topic: 'Autot',
          id: '1',
          selected: true,
          subtopics: [{
            topic: 'Virittely ja korjaaminen',
            id: '2',
            selected: false
          }]
        }],
        error: undefined
      },
      education: {
        data: {
          unverifiedEducations: [],
          selection: undefined
        }
      }
    },
    value: {
      profile: {
        interests: queryStatus
      }
    }
  })

  return {
    Context: React.createContext(state)
  }
})

describe('Interests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should show loading indicator when fetching suggestions', () => {
    let Interests
    mockState('pending')
    jest.isolateModules(() => { Interests = require('./Interests').default })

    const renderedJSON = createRendererWithTheme(<Interests/>).toJSON()
    expect(renderedJSON).toMatchSnapshot()
  })

  it('should show interest suggestions when fetched successfully', () => {
    let Interests
    mockState('success')
    jest.isolateModules(() => { Interests = require('./Interests').default })

    const renderedJSON = createRendererWithTheme(<Interests/>).toJSON()
    expect(renderedJSON).toMatchSnapshot()
  })

  it('should show error when fetching interest suggestions failed', () => {
    let Interests
    mockState('failure')
    jest.isolateModules(() => { Interests = require('./Interests').default })

    const renderedJSON = createRendererWithTheme(<Interests/>).toJSON()
    expect(renderedJSON).toMatchSnapshot()
  })
})
