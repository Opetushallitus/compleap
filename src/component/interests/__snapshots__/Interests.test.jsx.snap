// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interests should show error when fetching interest suggestions failed 1`] = `
Array [
  <h1
    className="sc-cMljjf ghKTLH"
  >
    Kiinnostukset
  </h1>,
  <div>
    error
  </div>,
]
`;

exports[`Interests should show interest suggestions when fetched successfully 1`] = `
Array [
  <h1
    className="sc-kAzzGY fPsSYY"
  >
    Kiinnostukset
  </h1>,
  <p>
    Valitse mahdollisimman tarkasti mitkä allaolevista asioista sinua kiinnostavat. Mitä tarkemmin valitset, sitä tarkemmin pystymme etsimään sinulle sopivia koulutuksia.
  </p>,
  <h2
    className="sc-chPdSV bVBjBE"
  >
    Valitse seuraavista ainakin 5 sinua kiinnostavaa asiaa
  </h2>,
  <ul
    className="sc-jWBwVP krmyvL"
  >
    <li
      className="sc-brqgnP lcnAMd"
    >
      <div
        style={
          Object {
            "display": "contents",
          }
        }
      >
        <button
          className="sc-kpOJdX jsGued"
          onClick={[Function]}
          selected={true}
          value="Autot"
        >
          Autot
          <div
            className="sc-dxgOiQ hhNrT"
            selected={true}
          >
            <svg
              data-file-name="SvgCheckmark"
              style={
                Object {
                  "height": "1rem",
                  "marginTop": "0.7rem",
                  "stroke": "#ffffff",
                  "strokeWidth": 2,
                  "width": "1rem",
                }
              }
            />
          </div>
        </button>
      </div>
      <ul
        className="sc-ckVGcZ iUctNF"
      >
        <li
          className="sc-jKJlTe ivktGD"
        >
          <div
            style={
              Object {
                "display": "contents",
              }
            }
          >
            <button
              className="sc-kpOJdX iIMEAZ"
              onClick={[Function]}
              selected={false}
              value="Virittely ja korjaaminen"
            >
              Virittely ja korjaaminen
              <div
                className="sc-dxgOiQ laCMQ"
                selected={false}
              >
                <svg
                  data-file-name="SvgCheckmark"
                  style={
                    Object {
                      "height": "1rem",
                      "marginTop": "0.7rem",
                      "stroke": "#e2e2e2",
                      "strokeWidth": 2,
                      "width": "1rem",
                    }
                  }
                />
              </div>
            </button>
          </div>
        </li>
      </ul>
    </li>
  </ul>,
]
`;

exports[`Interests should show loading indicator (placeholder tags) when fetching suggestions 1`] = `
Array [
  <h1
    className="sc-bdVaJa hqCxXu"
  >
    Kiinnostukset
  </h1>,
  <div
    className="sc-iwsKbI laYIiJ"
  >
    <div
      className="sc-VigVT TyGGh"
    >
      <div
        className="sc-htoDjs cKmTuq"
      >
        <div
          className="sc-dnqmqq jcnWGu"
        />
        <div
          className="sc-dnqmqq jcnWGu"
        />
        <div
          className="sc-dnqmqq jcnWGu"
        />
        <div
          className="sc-dnqmqq jcnWGu"
        />
      </div>
    </div>
    <div
      className="sc-gZMcBi bIDViS"
    >
      <div
        className="sc-fjdhpX khYAnD"
        width="55%"
      />
      <div
        className="sc-fjdhpX hABshK"
        width="77%"
      />
      <div
        className="sc-fjdhpX ezCgPo"
        width="65%"
      />
    </div>
    <div
      className="sc-gqjmRU ksnVNk"
    >
      <div
        className="sc-jTzLTM hDugnc"
        width="2rem"
      />
      <div
        className="sc-jTzLTM eRznqm"
        width="3rem"
      />
      <div
        className="sc-jTzLTM jtDwYV"
        width="5rem"
      />
      <div
        className="sc-jTzLTM bYVLcY"
        width="4rem"
      />
      <div
        className="sc-jTzLTM iIkCPF"
        width="3rem"
      />
      <div
        className="sc-jTzLTM UOrTg"
        width="2rem"
      />
      <div
        className="sc-jTzLTM bYVLcY"
        width="4rem"
      />
      <div
        className="sc-jTzLTM eRznqm"
        width="3rem"
      />
      <div
        className="sc-jTzLTM jtDwYV"
        width="5rem"
      />
      <div
        className="sc-jTzLTM kfvbdv"
        width="4rem"
      />
      <div
        className="sc-jTzLTM UOrTg"
        width="2rem"
      />
      <div
        className="sc-jTzLTM eRznqm"
        width="3rem"
      />
      <div
        className="sc-jTzLTM jtDwYV"
        width="5rem"
      />
      <div
        className="sc-jTzLTM bYVLcY"
        width="4rem"
      />
      <div
        className="sc-jTzLTM eRznqm"
        width="3rem"
      />
      <div
        className="sc-jTzLTM bYVLcY"
        width="4rem"
      />
      <div
        className="sc-jTzLTM iIkCPF"
        width="3rem"
      />
      <div
        className="sc-jTzLTM bYVLcY"
        width="4rem"
      />
      <div
        className="sc-jTzLTM eRznqm"
        width="3rem"
      />
      <div
        className="sc-jTzLTM UOrTg"
        width="2rem"
      />
      <div
        className="sc-jTzLTM jtDwYV"
        width="5rem"
      />
      <div
        className="sc-jTzLTM bYVLcY"
        width="4rem"
      />
    </div>
  </div>,
]
`;
